### ISSUE TITLE
search for address, having multiple caching providers active, gives results hundreds of km away

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#7279]
##### Detailed steps causing the problem:
- I have gc.com, oc.de and oc.nl enabled
- when I do a search for address, I get 20 results for each provider
- some of these results are hundreds of km away

##### Actual behavior after performing these steps:
- oc.nl sends me some result at the border of Germany & Netherlands
- although I'm doing a search around, lets say, Cologne or Munich

##### Expected behavior after performing these steps:
- feature request: skip results which are more than x <variable setting> km away

##### Version of c:geo used:
- last release and current NB

##### Is the problem reproducible for you?
Yes

**Comment by Lineflyer** [comment]
Well, there is already a distance limit implemented AFAIK. Maybe it can be reduced a bit (or kind of restrict the shown results to a mean value of the distances of all results returned by all connectors)

**Comment by moving-bits** [comment]
Ok, did some research in the sourcecode: The oc connector actually has a max. radius limitation, which is set to 200 km currently. If I change this to 5 km and search around Cologne, no oc.nl cache gets listed. => works

So maybe I should just make this setting user configurable...

**Comment by bekuno** [comment]
and 20km as default? :-)

**Comment by Lineflyer** [comment]
Addition:
Given that gc.com caches have highest density among our connectors (at least before geocaching.su will be included) in most parts of the world, we could maybe use the something like maxrangeOC=maxdistancesearchresultGC-mindistancesearchresultGC   
Additionally we need to think about what should happen if the user selects "Load more caches".

**Comment by moving-bits** [comment]
When doing my quick glance at the source I could not find a radius limit for gc.com connector, only for oc connector (have not checked any other).

_(Oops, can't update my post fast enough while you are adding more content to yours :-))_

To stay neutral in terms of connectors I would stick with a user configurable setting - this would respect different cache densities in different parts of the earth as well. We could either put it in the settings part as suggested in my previous post, or make it accessible through the dot menu during address based search.

**Comment by Lineflyer** [comment]
> As for the "addition" post I'm not really convinced by the suggestion. Some of the drawbacks you've already mentioned. Also: Wouldn't a formula like maxrangeOC=maxdistancesearchresultGC imply that we would have to call the gc connector first, wait for the results, and only then could call the other connectors? To my understanding all connectors currently get called in parallel.

Yeah, thats right. It should also work for users only using other connector(s) than gc.com, which would be not the case in your example.

> To stay neutral in terms of connectors and to avoid introducing a strict calling sequence I would prefer to stick with a user configurable setting - this would respect different cache densities in different parts of the earth as well. We could either put it in the settings part as suggested in my previous post, or make it accessible through the dot menu during address based search.

I could be user configurable, appreciate that.
But the default value is kind of a discussion.
As this query is also the base for "Nearby search" I would set it much lower than 200km, maybe to 20km as what is usually reachable with a short drive by car.

**Comment by moving-bits** [comment]
> I could be user configurable, appreciate that.
> But the default value is kind of a discussion.
> As this query is also the base for "Nearby search" I would set it much lower than 200km, maybe to 20km as what is usually reachable with a short drive by car.

200 km is what currently is used by the OKAPI based connectors (= opencaching.xxx platforms), that why I chose this as default value. But I'm open to 20 km as well.

OTOH - would this make such a big difference? Nearby search also stops after 20 founds IIRC, so the radius probably never gets to be the limit for dense populated areas. And for other areas - "nearby" is relative :-)

**Comment by Lineflyer** [comment]
> Nearby search also stops after 20 founds IIRC,

This depends on the site I assume...just now testing gc.su which returns 720 results for address search "Moscow" which is fine as they are in a small radius around Moscow (high density).

**Comment by rsudev** [comment]
I don't think a user-configurable limit is the solution to the problem. A more flexible handling of storing or mapping the search results, that takes the possibility of such outliers into account might be a better approach. And if the default is much lower than the current setting, I would assume that it is bound to fail users in other parts of the world, where the cache density is different. Because 99% will never even look for such a setting IMO...

**Comment by okainov** [comment]
>Because 99% will never even look for such a setting IMO...

This is very true, I have to admit :)

**Comment by Lineflyer** [comment]
History of related issues/change FYI:
#3542
#3211
#3227

**Comment by moving-bits** [comment]
@Lineflyer Thanks for the additional links. They describe the problem (and a possible solution) quite accurate. But with that the fix will be much more (how was it called?) _interesting_... :-)

**Comment by murggel** [comment]
Asked on c:geo-booth at Event [Tanz auf den Vulkanen](https://coord.info/GCATPRH)
Probably the setting for the limitation was not known by the user?
